src/MultiCollateralVault.sol

```diff
@@ -0,0 +1,275 @@
+        uint256 ltvRatio_ = WDiv.wdiv(borrowingPower_, debt[user]);
+        uint256 maxRatio_ = WDiv.wdiv(borrowingPower_, depositsInETH_);
+        bool collateralized;
+        if(ltvRatio_ <= maxRatio_){
+            collateralized = true;
+        } else {
+            collateralized = false;
+        }
+        return collateralized;
+    }
+/***
+-------------------------------------------------------------------------------------------------------------------------------------------------------------
+    User Functions
+-------------------------------------------------------------------------------------------------------------------------------------------------------------
+ */
+    
+    /// @notice allows a user to deposit WETH for use as collateral
+    /// @dev to ensure that a token is being accepted as collateral we check to ensure that a maximum loan to value ratio has been set since
+    /// that is the final step in adding a new token in collateral
+    function deposit(IERC20 token, uint256 amount) external {
+        require(collateralInfos[token].ratio != 0, "token not currently accepted as collateral");
+        deposits[msg.sender][token] += amount;
+        token.safeTransferFrom(msg.sender, address(this), amount);
+        emit Deposit(msg.sender, token, amount);
+    }
+
+    /// @notice allows a user to borrow WETH
+    function borrow(uint256 amount) external {
+        require(amount <= _remainingPower(msg.sender), "Insufficient collateral");
+        debt[msg.sender] += amount;
+        WETH.safeTransfer(msg.sender, amount);
+        emit Borrow(msg.sender, amount);
+    }
+
+    /// @notice allows a user to withdraw token that are not currently utilized as collateral
+    function withdraw(IERC20 token, uint256 amount) external {
+        require(amount < _freeCollateral(msg.sender, token), "Position would liquidate");
+        deposits[msg.sender][token] -= amount;
+        token.safeTransfer(msg.sender, amount);
+        emit Withdraw(msg.sender, token, amount);
+    }
+
+    /// @notice allows a user to repay their debts
+    function repay(uint256 amount) external {
+        require(amount <=  debt[msg.sender], "repayment exceeds debt");
+        debt[msg.sender] -= amount;
+        IERC20(WETH).safeTransferFrom(msg.sender, address(this), amount);
+        emit Repay(msg.sender, amount,  debt[msg.sender]);
+    }
+
+/***
+-------------------------------------------------------------------------------------------------------------------------------------------------------------
+    Helper Functions
+-------------------------------------------------------------------------------------------------------------------------------------------------------------
+ */
+
+    /// @dev returns the number of decimals less than 18 a token uses (ie USDC has 6 decimals returns 12)
+    function _decimalsDifference(IERC20 token) internal view returns(uint8){
```

Comment:
Once you remove `_remainingPower` and `_freeCollateral` you can just inline this on `_borrowingPower` and remove another function.